core/src/main/java/io/questdb/griffin/engine/table/LatestByValuesFilteredRecordCursor.java

```diff
@@ -59,8 +60,9 @@ public void toTop() {
     }
 
     @Override
-    protected void buildTreeMap(SqlExecutionContext executionContext) {
+    protected void buildTreeMap(SqlExecutionContext executionContext) throws SqlException {
         prepare();
+        filter.init(this, executionContext);
```

Comment:
Why 
```
int frameColumnIndex = columnIndexes.getQuick(columnIndex);
```
is not needed here?

Reply:
That's because we need to initialize the filter before accessing it (see the very first bug in the PR description).

If you comment out this line of code and revert the change in `LatestByValuesFilteredRecordCursorFactory`, you'd get such an NPE in `testLatestByFilteredBySymbolInNoIndexes`:
```
java.lang.NullPointerException
	at io.questdb/io.questdb.griffin.engine.functions.bool.InSymbolFunctionFactory$Func.getBool(InSymbolFunctionFactory.java:110)
	at io.questdb/io.questdb.griffin.engine.table.LatestByValuesFilteredRecordCursor.buildTreeMap(LatestByValuesFilteredRecordCursor.java:76)
	at io.questdb/io.questdb.griffin.engine.table.AbstractDescendingRecordListCursor.of(AbstractDescendingRecordListCursor.java:73)
	at io.questdb/io.questdb.griffin.engine.table.AbstractTreeSetRecordCursorFactory.getCursorInstance(AbstractTreeSetRecordCursorFactory.java:60)
	at io.questdb/io.questdb.griffin.engine.table.AbstractDeferredTreeSetRecordCursorFactory.getCursorInstance(AbstractDeferredTreeSetRecordCursorFactory.java:103)
	at io.questdb/io.questdb.griffin.engine.table.LatestByValuesFilteredRecordCursorFactory.getCursorInstance(LatestByValuesFilteredRecordCursorFactory.java:83)
	at io.questdb/io.questdb.griffin.engine.table.LatestByValuesFilteredRecordCursorFactory.getCursorInstance(LatestByValuesFilteredRecordCursorFactory.java:41)
	at io.questdb/io.questdb.griffin.engine.table.AbstractDataFrameRecordCursorFactory.getCursor(AbstractDataFrameRecordCursorFactory.java:47)
	at io.questdb/io.questdb.griffin.engine.table.AbstractDeferredTreeSetRecordCursorFactory.getCursor(AbstractDeferredTreeSetRecordCursorFactory.java:39)
	at io.questdb/io.questdb.griffin.engine.groupby.GroupByRecordCursorFactory.getCursor(GroupByRecordCursorFactory.java:89)
	at io.questdb/io.questdb.test.tools.TestUtils.printSql(TestUtils.java:674)
	at io.questdb/io.questdb.test.tools.TestUtils.assertSql(TestUtils.java:436)
	at io.questdb/io.questdb.griffin.SqlCodeGeneratorTest.lambda$testLatestByFilteredBySymbolIn$48(SqlCodeGeneratorTest.java:4328)
```

What happens here is that `InSymbolFunction` tries to access its private test function which wasn't yet initialized. We didn't see this NPE before because there were no tests for this latest by + filter combination.

Reply:
I don't disagree with filter init. My question is why column index is not remapped here.

Reply:
Sorry, I misunderstood the question. Let me check that.

Reply:
This method deals with `TableReaderSelectedColumnRecord` class which remaps select column indexes to table ones. So, in the `recordA.getInt(columnIndex)` call there is a `deferenceColumn()` method call which does the remapping. That's why we don't need to do extra index remapping here.